pkg/k8s/apis/cilium.io/v2/client/register.go

```diff
@@ -503,11 +503,7 @@ func createUpdateCRD(clientset apiextensionsclient.Interface, CRDName string, cr
 		return false, err
 	})
 	if err != nil {
-		deleteErr := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(context.TODO(), crd.ObjectMeta.Name, metav1.DeleteOptions{})
-		if deleteErr != nil {
-			return fmt.Errorf("unable to delete k8s %s CRD %s. Deleting CRD due: %s", CRDName, deleteErr, err)
-		}
-		return err
+		return fmt.Errorf("error occurred waiting for CRD: %w", err)
```

Comment:
This is correct. 

What's missing is the removal of the code in `operator/crd_test.go` as seen in the upstream commit.